### PR TITLE
[Scheduler] have `TriggerInterface` extend `\Stringable`

### PR DESCRIPTION
**Pull request by kbond**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | n/a
| License       | MIT
| Doc PR        | todo

Helps with the output of `debug:schedule` and with #49838.

### CODE REVIEW

File: src/Symfony/Component/Scheduler/Trigger/DatePeriodTrigger.php [hunk 1]
```diff
@@ -21,6 +21,16 @@ public function __construct(
     ) {
     }
 
+    public function __toString(): string
+    {
+        return \sprintf(
+            '%s - %s, %s',
+            $this->period->getStartDate()->format(\DateTimeInterface::ATOM),
+            $this->period->getEndDate()?->format(\DateTimeInterface::ATOM) ?? '(inf)',
+            'every ?',
```
Comment (kbond) [inline src/Symfony/Component/Scheduler/Trigger/DatePeriodTrigger.php]: Not sure what to put here but if #49817 is merged, this won't be necessary.

File: src/Symfony/Component/Scheduler/Trigger/CallbackTrigger.php [hunk 2]
```diff
@@ -20,11 +20,16 @@ final class CallbackTrigger implements TriggerInterface
 {
     private \Closure $callback;
 
-    public function __construct(callable $callback)
+    public function __construct(callable $callback, private string $description)
```
Comment (fabpot) [inline src/Symfony/Component/Scheduler/Trigger/CallbackTrigger.php]: I would not make it required. If not provided, let's use `spl_object_hash($tis->callback)`.

File: src/Symfony/Component/Scheduler/Trigger/DatePeriodTrigger.php (hunk 1 above)
Comment (fabpot) [reply src/Symfony/Component/Scheduler/Trigger/DatePeriodTrigger.php]: Let's use `-` for now to allow this PR to be merged quickly (#49817 will be merged before 6.3 final, we just need to figure out what we want to do there).

**Review by fabpot — CHANGES_REQUESTED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @kbond.

**Pull request merged**